rational/src/lib.rs

```diff
@@ -774,6 +932,15 @@ mod test {
         let _a = Ratio::new(1, 0);
     }
 
+    #[test]
+    fn test_approximate_float() {
+        assert_eq!(super::approximate_float(0.5f32, 1.0e-20, 30), Some(Ratio::new(1i64, 2)));
+        assert_eq!(super::approximate_float(0.5f64, 1.0e-20, 30), Some(Ratio::new(1i32, 2)));
+        assert_eq!(super::approximate_float(5f32, 1.0e-20, 30), Some(Ratio::new(5i64, 1)));
+        assert_eq!(super::approximate_float(5f64, 1.0e-20, 30), Some(Ratio::new(5i32, 1)));
+        assert_eq!(super::approximate_float(29.97f32, 1.0e-20, 30), Some(Ratio::new(2997i64, 100)));
+        assert_eq!(super::approximate_float(-29.97f32, 1.0e-20, 30), Some(Ratio::new(-2997i64, 100)));
```

Comment:
How about some floating point boundaries?  NaN, infinity, and even the finite yet absurdly large and small.  And unsigned ratios need testing now too, including that behavior with negative floats.

I'll bet we could benefit from quickcheck throughout num!